From 745679ae193b8a9731dd5e34ad3fcd68e23da7b1 Mon Sep 17 00:00:00 2001
From: Carl Smedstad <carl.smedstad@protonmail.com>
Date: Fri, 23 Aug 2024 07:17:47 +0200
Subject: [PATCH 2/2] Exclude tests from wheel

As the wheel's primary purpose is to be installed in a virtualenv or on a
system, the tests for testrunner itself are unnecessary to include. Including
them also makes distro-tooling such as Arch Linux's namcap produce noisy
warnings regarding missing dependencies.

The tests are still included in the sdist as per the MANIFEST.in file.
---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 44822fc..7f8d656 100644
--- a/setup.py
+++ b/setup.py
@@ -91,7 +91,7 @@ setup(
     long_description=long_description,
     author='Zope Foundation and Contributors',
     author_email='zope-dev@zope.dev',
-    packages=find_namespace_packages(where='src'),
+    packages=find_namespace_packages(where='src', exclude=['zope.testrunner.tests*']),
     package_dir={'': 'src'},
     classifiers=[
         "Development Status :: 5 - Production/Stable",
@@ -124,6 +124,6 @@ setup(
         'distutils.commands': [
             'ftest = zope.testrunner.eggsupport:ftest'],
     },
-    include_package_data=True,
+    include_package_data=False,
     zip_safe=False,
 )
-- 
2.46.0

